pandas/core/ops/common.py

```diff
@@ -70,10 +69,10 @@ def new_method(self, other):
             # For comparison ops, Index does *not* defer to Series
             pass
         else:
-            for cls in [ABCDataFrame, ABCSeries, ABCIndex]:
-                if isinstance(self, cls):
-                    break
-                if isinstance(other, cls):
+            prio = getattr(other, "__pandas_priority__", None)
+            if prio is not None:
+                if prio > self.__pandas_priority__:
```

Comment:
Should a warning be raised if `prior == self.__pandas_priority__`? It may indicate someone was trying to set the prior to a high-enough-round-number expecting something to happen or pandas objects may share the same priority?

Reply:
id hold off on this until there's an indication this is a pain point